docs/docs/building-modules/09-autocli.md

```diff
@@ -0,0 +1,129 @@
+---
+sidebar_position: 1
+---
+
+
+# AutoCLI
+
+:::note Synopsis
+This document details how to build CLI and REST interfaces for a module. Examples from various Cosmos SDK modules are included.
+:::
+
+:::note
+
+## Pre-requisite Readings
+
+* [Building Modules Intro](./01-intro.md)
+
+:::
+
+The `autocli` package is a [Go library](https://pkg.go.dev/cosmossdk.io/client/v2/autocli) for generating CLI (command line interface) interfaces for Cosmos SDK-based applications. It provides a simple way to add CLI commands to your application by generating them automatically based on your gRPC service definitions. Autocli generates CLI commands and flags directly from your protobuf messages, including options, input parameters, and output parameters. This means that you can easily add a CLI interface to your application without having to manually create and manage commands.
+
+## Getting Started
+
+Here are the steps to use the `autocli` package:
+
+1.  Define your app's modules that implement the `appmodule.AppModule` interface.
+2.  Create an instance of the `autocli.AppOptions` struct that specifies the modules you defined. 
```

Comment:
I would explain it is a method on app module `func (am AppModule) AutoCLIOptions() *autocliv1.ModuleOptions`